File: rust_exercises/leetcode/Problem748/Solution.java
class Solution {
    public String shortestCompletingWord(String licensePlate, String[] words) {
        int[] licenseMap = new int[26];
        for (int i = 0; i < licensePlate.length(); i += 1) {
            char ch = licensePlate.charAt(i);
            if (Character.isAlphabetic(ch)) {
                int index = (int) (Character.toLowerCase(ch) - 'a');
                licenseMap[index] += 1;
            }
        }
        
        String shortest = null;
        for (String word: words) {
            int[] wordMap = new int[26];
            for (int i = 0; i < word.length(); i += 1) {
                char ch = word.charAt(i);
                int index = (int) (ch - 'a');
                wordMap[index] += 1;
            }
            int i;
            for (i = 0; i < 26; i += 1) {
                if (wordMap[i] < licenseMap[i]) {
                    break;
                }
            }
            if (i >= 26 && (shortest == null || word.length() < shortest.length())) {
                shortest = word;
            }
        }
        
        return shortest;
    }
}

File: rust_exercises/leetcode/Problem748/solution.rs
impl Solution {
    pub fn shortest_completing_word(license_plate: String, words: Vec<String>) -> String {
        let mut license_map = vec![0; 26];
        for ch in license_plate.chars() {
            if ch.is_alphabetic() {
                let index = ch.to_lowercase().next().unwrap() as usize - 'a' as usize;
                license_map[index] = license_map[index] + 1;
            }
        }
        
        let mut shortest: Option<&String> = None;
        
        for word in words.iter() {
            let mut word_map = vec![0; 26];
            for ch in word.chars() {
                let index = ch as usize - 'a' as usize;
                word_map[index] = word_map[index] + 1;
            }
            
            let mut i = 0;
            while i < 26 {
                if word_map[i] < license_map[i] {
                    break;
                }
                i += 1;
            }
            if (i >= 26 && (shortest.is_none() || word.len() < shortest.unwrap().len())) {
                shortest = Some(word);
            }
        }
        
        shortest.unwrap().clone()
    }
}
